### PR TITLE
Fix/modify mealrepo to use locald db

### PR DESCRIPTION
**Pull request by RandomUsername1315**
# Pull Request Overview

This simple pull request modifies the `MealViewModel` to use `MealRepository` instead of `FirebaseMealRepository`. This change ensures that data is cached in the local database and fetched from the Firebase database only when necessary.

## Changes

- **Modify `MealViewModel.kt` Class**: Updated the previous implementation to use the repository, which decides whether a request should be made locally or to the Firebase database.
- **Modify `MainActivity.kt` and `AddMealFlow.kt` Classes**: Updated to handle the new implementation of the ViewModel.

## Notes

- #126 was approved but did not merge correctly with the main branch. When #114 was approved for merging, it did not directly target the main branch; instead, it targeted the branch that created the local database for meals (which had been merged first). When #126 was approved and merged, the other branch was reinstated. Causing the need for this PR, even if all changes were already approved. 
- This change also makes use of Hilt Di for a cleaner implementation

### CODE REVIEW

File: app/src/main/java/com/github/se/polyfit/viewmodel/meal/MealViewModel.kt [hunk 1]
```diff
@@ -30,30 +32,70 @@ constructor(private val user: User, private val mealRepo: MealFirebaseRepository
...
+            mealTemp,
+            nutritionalInformation,
+            ingredients,
+            firebaseID,
+            createdAt)
+
+    _isComplete.value = _meal.value?.isComplete() ?: false
   }
 
   fun setMealName(name: String) {
```
Comment (aprilslis) [inline app/src/main/java/com/github/se/polyfit/viewmodel/meal/MealViewModel.kt:93]: Now that we have `setMealData`, `setMealName` is not really needed, we should maybe remove it and just use `setMealData` for all name-changing place

File: app/src/androidTest/java/com/github/se/polyfit/viewmodel/MealViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    io.mockk.clearMocks(mealRepo)
+  }
+
+  @Test
+  fun testInitialization_withFirebaseID_loadsData() {
+    assert(viewModel.meal.value?.name == "Test Meal")
+  }
+
+  @Test
+  fun testSetMealName_updatesMealName() {
```
Comment (aprilslis) [inline app/src/androidTest/java/com/github/se/polyfit/viewmodel/MealViewModelTest.kt:68]: this test will no longer be needed if you remove `setMealName` and use `setMealData` only

File: app/src/androidTest/java/com/github/se/polyfit/MainActivityTest.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import androidx.test.filters.LargeTest
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class MainActivityTest {
+
+  @Test
+  fun testActivityShouldBeLaunched() {
```
Comment (aprilslis) [inline app/src/androidTest/java/com/github/se/polyfit/MainActivityTest.kt:14]: Why is this empty test here? MainActivity is not changed, nor the test is testing for anything?

**Review by aprilslis — APPROVED** [review]
Looks overall! after you resolve conflicts and comment it would be good!

File: app/src/androidTest/java/com/github/se/polyfit/MainActivityTest.kt (hunk 3 above)
Comment (RandomUsername1315) [reply app/src/androidTest/java/com/github/se/polyfit/MainActivityTest.kt:14]: Haha, I don't know how that got there. It's definitely a mistake. Thanks for catching it.

**Pull request merged**